src/Raven.Server/Documents/Indexes/Persistence/IndexOperationBase.cs

```diff
@@ -119,25 +120,24 @@ protected static int LuceneGetPageSize(IndexSearcher searcher, long pageSize)
     protected static int CoraxGetPageSize(global::Corax.IndexSearcher searcher, int pageSize, IndexQueryServerSide query, bool isBoolean = false)
     {
         var numberOfEntries = searcher.NumberOfEntries;
-        
         if (numberOfEntries == 0)
             return 16;
         
-        //If we have a binary operation, we need to pass a buffer large enough to hold all the individual results.
-        //We need to do this to get correct results and since we don't know how much results subqueries will have  we must create a buffer big enough to get all items from index
+        // If we have a binary operation, we need to pass a buffer large enough to hold all the individual results.
+        // We need to do this to get correct results and since we don't know how much results subqueries will have  we must create a buffer big enough to get all items from index
         if (query.Metadata.OrderBy is not null || query.Metadata.IsDistinct || isBoolean)
         {
-            if (numberOfEntries > int.MaxValue)
-                return int.MaxValue;
+            if (numberOfEntries > MaxBufferSizeForCorax)
+                return MaxBufferSizeForCorax;
```

Comment:
How is asking for 128K results handled, then?

Reply:
Much better. Haven't seen any issue so far. The problem was also the different sizes caused the ArrayPool to have to keep arrays of all sizes. And the real issue happen on a few very big requests that ended up in the large objects heap.